File: backend/src/main/java/io/github/artemy/osipov/shop/service/candy/Candy.java
package io.github.artemy.osipov.shop.service.candy;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.util.UUID;

@Data
@Document("candies")
public class Candy {

    @Id
    private UUID id;
    private String name;
    private String firm;
    private BigDecimal price;
    private Double order;
    private Boolean active = true;

    public void hide() {
        active = false;
    }
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/service/candy/Candy.kt
package io.github.artemy.osipov.shop.service.candy

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.math.BigDecimal
import java.util.*

@Table("candies")
data class Candy(
    @Id
    val id: UUID,
    var name: String,
    var firm: String,
    var price: BigDecimal,
    var order: Double,
    var active: Boolean = true
) {
    fun hide() {
        active = false
    }
}